ft-register-vendor():

Shumba250 opened the pull request:
#### What does this PR do?
Register Vendors
documented with swagger
#### Description of Task to be completed

- the administrator should be able to register a new vendor
-  an email containing a random password should be sent to the provided email 
- the vendor should be able to sign in using the provided email and password
- appropriate error messages are displayed on unsuccessful attempts
- the tests are passing
- documented with swagger
#### How should this be manually tested?

- clone the repo
- use npm install to install all packages
- use  npx sequelize db:migrate to migrate migrations
- use npm run dev to start the server
- use the appropriate router in postman or thunderclient  forexample localhost5050/example
- you should receive a random password on the provided email or
- navigate to localhost/5050/docs and test it using the documented API, the result should be the same
- use npm test to test the endpoint

Shumba250 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Register vendors](https://trello.com/c/fYQhil84/4-register-vendors)

sostene12 reviewed (commented):
- solve the conflicts, also if you would split the docs files, to be readable very clearly

wakaflorien commented on test/user.test.js:
```diff
@@ -8,14 +8,15 @@ describe("testing all routes", () => {
-			lastName: "helen",
-			email: "user@example.com",
+
+describe("vendor", ()=> {
```
- test for post vendor is failing
- make sure you test more scenarios not just `post` successful. test failing scenarios too (validation, badrequest, authorization) 
- improve description to howto manually test this. specifically image type to submit.

wakaflorien commented on src/routes/index.js:
```diff
@@ -1,8 +1,8 @@
 const routes = express.Router();
 
-routes.use("/", route);
+routes.use("/", vendorRoute);
```
- it would be better not to use vendor route on home route.

wakaflorien reviewed (changes requested):
- time out when i run tests.

sostene12 commented:
- i think you can refactor a little bit of your codes to be reusable,
-  also if we can use class based codes to declare more than one fuction it can be better
-  the related endpoints can have their doc file(swagger)

You did a great job

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

wakaflorien commented on test/user.test.js:
```diff
@@ -9,13 +9,35 @@ describe("testing all routes", () => {
+    const response = await request(app).post("/signup").send({
+      firstName: "umurungi",
+      lastName: "helen",
+      email: "mudakikwaaimable05@gmail.com"
```
- what happens when the user is not already there?

patsicko reviewed (commented):
Can we have the same model to avoid some conficts? also consider to add all test variables required for test.
For validation, can we have one reusable middleware that validates all schema than just having a midleware for every schema?

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

patsicko commented on package.json:
```diff
@@ -20,7 +20,7 @@
     "db:seed": "sequelize-cli seed:generate --name",
     "db:seeds": "sequelize-cli db:seed:all",
     "db:seed:undo": "sequelize-cli db:seed:undo",
-    "test":"cross-env NODE_ENV=test npx sequelize db:migrate:undo:all && cross-env NODE_ENV=test  npm run migrate:up -all && jest --detectOpenHandles --coverage --testTimeout 100000 --forceExit",
```
Consider not running db:migrate:undo:all before tests, that may be the reason why your tests are failing because you may be deleting some migrations that are not created. you can use this script instead   
"test": "cross-env NODE_ENV=test  npm run migrate:up -all && jest --detectOpenHandles --coverage --testTimeout 100000 --forceExit"

patsicko reviewed (changes requested):
Consider not running db:migrate:undo:all before tests, that may be the reason why your tests are failing because you may be deleting some migrations that are not created. you can use this script instead
"test": "cross-env NODE_ENV=test npm run migrate:up -all && jest --detectOpenHandles --coverage --testTimeout 100000 --forceExit"

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts
